src/sparseml/pytorch/torchvision/export_onnx.py

```diff
@@ -0,0 +1,212 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import os
+from pathlib import Path
+from typing import Dict, Optional, Union
+
+import torch
+import torchvision
+from torch.nn import Module
+from torch.utils.data import DataLoader
+from torchvision.transforms.functional import InterpolationMode
+from tqdm import tqdm
+
+import click
+from sparseml.pytorch.models.registry import ModelRegistry
+from sparseml.pytorch.torchvision import presets
+from sparseml.pytorch.utils import ModuleExporter
+from sparseml.pytorch.utils.exporter import DEFAULT_ONNX_OPSET
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@click.command(
+    context_settings=dict(
+        token_normalize_func=lambda x: x.replace("-", "_"), show_default=True
+    )
+)
+@click.option(
+    "--arch-key",
+    type=str,
+    required=True,
+    help="The architecture key for image classification model; "
+    "example: `resnet50`, `mobilenet`. "
+    "Note: Will be read from the checkpoint if not specified",
+)
+@click.option(
+    "--checkpoint-path",
+    type=click.Path(dir_okay=False, file_okay=True, exists=True, path_type=Path),
+    required=True,
+    help="A path to a previous checkpoint to load the state from "
+    "and resume the state for exporting",
+)
+@click.option(
+    "--dataset-path",
+    type=click.Path(dir_okay=True, file_okay=False, exists=True, path_type=Path),
+    required=True,
+    help="The root dir path where the dataset is stored or should "
+    "be downloaded to if available",
+)
+@click.option(
+    "--labels-to-class-mapping",
+    type=click.Path(dir_okay=False, file_okay=True, exists=True, path_type=Path),
+    default=None,
+    help="Optional path to the dataset-specific mapping from "
+    "numeric labels to human-readable class strings. "
+    "Expected to be a path to a .json file containing "
+    "a serialized dictionary",
+)
+@click.option(
+    "--num-samples",
+    type=int,
+    default=-1,
+    help="The number of samples to export along with the model onnx "
+    "and pth files (sample inputs and labels as well as the "
+    "outputs from model execution)",
+)
+@click.option(
+    "--onnx-opset",
+    type=int,
+    default=DEFAULT_ONNX_OPSET,
+    help="The onnx opset to use for exporting the model",
+)
+@click.option(
+    "--save-dir",
+    type=click.Path(dir_okay=True, file_okay=False, path_type=Path),
+    default="torchvision",
+    help="The path to the directory for saving results",
+)
+@click.option(
+    "--convert-qat/--no-convert-qat",
+    default=True,
+    is_flag=True,
+    help="if True, exports of torch QAT graphs will be converted to a fully quantized "
+    "representation. Default is True",
+)
+@click.option(
+    "--interpolation",
+    default="bilinear",
+    type=str,
+    help="the interpolation method (default: bilinear)",
```

Comment:
defaults will be shown twice, cause `show_default=True` could we remove default from help message?

Reply:
Updated